src/runtime-rs/crates/hypervisor/src/ch/inner_device.rs

```diff
@@ -137,7 +164,85 @@ impl CloudHypervisorInner {
         Ok(())
     }
 
-    async fn handle_block_device(&mut self, cfg: BlockConfig) -> Result<()> {
+    async fn handle_vfio_device(&mut self, device: &VfioDevice) -> Result<()> {
+        // A device with multi-funtions, or a IOMMU group with one more
+        // devices, the Primary device is selected to be passed to VM.
+        // And the the first one is Primary device.
+        // safe here, devices is not empty.
+        let primary_device = device
+            .devices
+            .first()
+            .ok_or(anyhow!("Primary device list empty for vfio device"))?;
```

Comment:
Is it worth adding `device` to the error here?

```suggestion
            .ok_or(anyhow!("Primary device list empty for vfio device {:?}", device))?;
```